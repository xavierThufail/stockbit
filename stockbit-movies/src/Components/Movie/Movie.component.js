import React from 'react';
import { Button } from 'antd';

import { TouchableOpacity } from '../index';
import styles from './Movie.style';
import config from './Movie.config';

const Movie = ({ Title, Year, imdbID, Type, Poster, onClick, handleOpenModal }) => {
  return (
    <div style={styles.container}>
      <TouchableOpacity onClick={handleOpenModal(Poster)}>
        <img alt='img' src={Poster} style={styles.imgBtn} />
      </TouchableOpacity>
      <TouchableOpacity onClick={onClick} style={styles.content}>
        <h2 style={styles.title}>{Title}</h2>
        <p>{Year}</p>
        <p>{Type}</p>
      </TouchableOpacity>
    </div>
  );
};

Movie.displayName = config.displayName;
Movie.propTypes = config.propTypes;

export default Movie;