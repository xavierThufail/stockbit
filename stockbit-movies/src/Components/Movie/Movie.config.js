import PropTypes from 'prop-types';

const propTypes = {
  Title: PropTypes.string.isRequired,
  Year: PropTypes.string.isRequired,
  imdbID: PropTypes.string.isRequired,
  Type: PropTypes.string.isRequired,
  Poster: PropTypes.string.isRequired,
  onClick: PropTypes.func.isRequired,
  handleOpenModal: PropTypes.func.isRequired
};

const displayName = 'Movie';

const config = {
  displayName,
  propTypes
};

export default config;
