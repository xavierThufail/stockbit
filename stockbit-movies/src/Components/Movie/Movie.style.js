const styles = {
  container: {
    display: 'flex',
    padding: '24px',
    width: '100%'
  },
  imgBtn: {
    width: 100,
    height: 150,
    backgroundPosition: 'center',
    backgroundSize: 'cover',
    color: 'transparent',
    borderRadius: '10px'
  },
  content: { marginLeft: '24px', width: '100%' },
  title: {
    color: '#8f8f8f',
    lineHeight: 1
  }
};

export default styles;
