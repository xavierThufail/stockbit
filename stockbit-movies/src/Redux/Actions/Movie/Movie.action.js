import axios from 'axios';

import CONSTANTS from '../../../Constants/reducers';

const baseUrl = 'http://www.omdbapi.com?';
const apiKey = 'ce169b84';

const setLoading = (payload) => ({
  type: CONSTANTS.movie.SET_LOADING,
  payload
});

const setError = (payload) => ({
  type: CONSTANTS.movie.SET_ERROR,
  payload
});

const setMovieList = (payload) => ({
  type: CONSTANTS.movie.SET_LIST_MOVIE,
  payload
});

const setMovieDetail = (payload) => ({
  type: CONSTANTS.movie.SET_MOVIE_DETAIL,
  payload
});

const setSearchTerm = (payload) => ({
  type: CONSTANTS.movie.SET_SEARCH_TERM,
  payload
});

const fetchData = (props, type = 'search') => {
  const params = {
    apikey: apiKey,
    ...props
  };

  return (dispatch) => {
    dispatch(setLoading(true));

    if(type === 'detail') dispatch(setMovieDetail({}));

    axios
      .get(baseUrl, { params })
      .then(({ data }) => {
        if(type === 'search') {
          dispatch(setMovieList(data));
        } else {
          dispatch(setMovieDetail(data))
        }
      })
      .catch(err => dispatch(setError(err)));

    dispatch(setLoading(false));
  }
};

const fetchDetail = (props) => {

}

const movieActions = {
  fetchData,
  setSearchTerm
}

export default movieActions;
