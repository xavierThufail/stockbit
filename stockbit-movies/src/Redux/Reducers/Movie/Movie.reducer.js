import CONSTANTS from '../../../Constants/reducers';

const initialReducers = {
  listMovies: [],
  movieDetail: {},
  favorites: [],
  loading: false,
  error: null,
  totalPage: 0,
  searchTerm: ''
};

const movieReducers = (state = initialReducers, actions) => {
  switch(actions.type) {
    case CONSTANTS.movie.SET_LIST_MOVIE:
      const totalPage = Math.floor(actions.payload.Search.length / 5);
      const newListMovie = [];

      for(let i = 1; i <= totalPage; i++) {
        newListMovie.push(actions.payload.Search.slice((i - 1) * 5, i * 5))
      };

      return {...state, listMovies: newListMovie, totalPage};
    case CONSTANTS.movie.SET_MOVIE_DETAIL:
      return {...state, movieDetail: actions.payload};
    case CONSTANTS.movie.ADD_FAVORITE:
      return {...state, favorites: [...state.favorites, actions.payload]};
    case CONSTANTS.movie.SET_SEARCH_TERM:
      return {...state, searchTerm: actions.payload}
    case CONSTANTS.movie.SET_LOADING:
      return {...state, loading: actions.payload};
    case CONSTANTS.movie.SET_ERROR:
      return {...state, error: actions.payload};
    default:
      return state;
  };
};

export default movieReducers;
