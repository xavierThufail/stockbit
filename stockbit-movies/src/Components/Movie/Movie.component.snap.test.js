import React from 'react';
import { render } from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';

import Movie from './Movie.component';

describe('Movie component snap test', () => {
  it('Render Movie, should match snapshot', () => {

    const { container } = render(<Movie Title='test' Year='test' imdbID='test' Type='test' Poster='test' onClick={() => {}} handleOpenModal={() => {}}  />);

    expect(container).toMatchSnapshot();
  });
});
