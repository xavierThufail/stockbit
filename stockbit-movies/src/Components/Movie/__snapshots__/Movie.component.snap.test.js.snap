// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Movie component snap test Render Movie, should match snapshot 1`] = `
<div>
  <div
    style="display: flex; padding: 24px; width: 100%;"
  >
    <div
      class="touchable"
    >
      <img
        alt="img"
        src="test"
        style="width: 100px; height: 150px; background-position: center; background-size: cover; color: transparent; border-radius: 10px;"
      />
    </div>
    <div
      class="touchable"
      style="margin-left: 24px; width: 100%;"
    >
      <h2
        style="color: rgb(143, 143, 143); line-height: 1;"
      >
        test
      </h2>
      <p>
        test
      </p>
      <p>
        test
      </p>
    </div>
  </div>
</div>
`;
